### PR TITLE
fix: set minimal supported Node.js version to 14

### PR DESCRIPTION
**Pull request by kettanaito**
Follow up to https://github.com/mswjs/msw/pull/1136

Configured CI so it runs tests/build on both Node.js 14 and 16.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,14 +9,17 @@ on:
 jobs:
   build:
     runs-on: macos-latest
+    strategy:
+      matrix:
+        node: [14.19.0, 16.14.0]
```
Comment (MichaelDeBoey) [inline .github/workflows/ci.yml]: You can just use the latest version instead of running on specific Node version
```suggestion
        node: [14, 16]
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kettanaito) [reply .github/workflows/ci.yml]: Good catch. I was wondering if it'd support it this way 👍

**Comment by kettanaito** [comment]
We need to update `interceptors` and perhaps other ecosystem packages to also allow Node.js 14.

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by baptisteArno** [comment]
Does this fix `msw@0.39.1: The engine "node" is incompatible with this module. Expected version ">=16.x". Got "14.18.1"`?

**Comment by kettanaito** [comment]
@baptisteArno, precisely.

**Comment by thedavidprice** [comment]
Thank you @kettanaito Count me in if ever there's a way I can reciprocate collaboration!

**Comment by kettanaito** [comment]
## Released: v0.40.0 🎉

This has been released in v0.40.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v0.40.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/0.40.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**